### PR TITLE
[1.4.x]Add fix issue with setting value to openAPI title section

### PR DESCRIPTION
**Pull request by lnash94**
Fix #https://github.com/ballerina-platform/openapi-tools/issues/1252

```ballerina
import ballerina/http;

@http:ServiceConfig {
    cors: {
        allowOrigins: ["*"]
    }
}

service / on new http:Listener(9090) {

    # A resource for generating greetings
    # + name - the input string name
    # + return - string name with hello message or error
    resource function get greeting(string name) returns string|error {
        // Send a response back to the caller.
        if name is "" {
            return error("name should not be empty!");
        }
        return "Hello, " + name;
    }
}
```
Generated OAS
```openapi
openapi: 3.0.1
info:
  title: "/"
  version: 0.0.0
servers:
  - url: "{server}:{port}/"
    variables:
      server:
        default: http://localhost
      port:
        default: "9090"
paths:
  /greeting:
    get:
      summary: A resource for generating greetings
      operationId: getGreeting
      parameters:
        - name: name
          in: query
          description: the input string name
          required: true
          schema:
            type: string
      responses:
        "200":
          description: Ok
          content:
            text/plain:
              schema:
                type: string
        "500":
          description: Internal server error
          content:
            text/plain:
              schema:
                type: string
components: {}
```
Expected 
Generated OAS
```openapi
openapi: 3.0.1
info:
  title: "<file name>"
  version: 0.0.0
servers:
  - url: "{server}:{port}/"
    variables:
      server:
        default: http://localhost
      port:
        default: "9090"
...
```

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: openapi-bal-service/src/main/java/io/ballerina/openapi/converter/utils/ServiceToOpenAPIConverterUtils.java [hunk 1]
```diff
@@ -290,23 +290,31 @@ private static OASResult fillOpenAPIInfoSection(ServiceDeclarationNode serviceNo
...
         } else {
-            openAPI.setInfo(new Info().version(version).title(normalizeTitle(currentServiceName)));
+            setInfoDetailsIfServiceNameAbsent(openapiFileName, openAPI, currentServiceName, version);
         }
 
         return new OASResult(openAPI, diagnostics);
     }
 
+    /**
+     * This function may fill the details for openAPI Info section when the service base path is absent or `/`.
```
Comment (NipunaRanasinghe) [inline openapi-bal-service/src/main/java/io/ballerina/openapi/converter/utils/ServiceToOpenAPIConverterUtils.java:307]:
```suggestion
     * Generates openAPI Info section when the service base path is absent or `/`.
```

File: openapi-cli/src/test/resources/ballerina-to-openapi/response/service_config_with_cors_with_slash.bal [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import ballerina/http;
+
+@http:ServiceConfig {
+    cors: {
+        allowOrigins: ["*"]
+    }
+}
+
```
Comment (NipunaRanasinghe) [inline openapi-cli/src/test/resources/ballerina-to-openapi/response/service_config_with_cors_with_slash.bal:8]: additional newline

File: openapi-cli/src/test/resources/ballerina-to-openapi/response/service_config_with_cors_without_base_path.bal [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import ballerina/http;
+
+@http:ServiceConfig {
+    cors: {
+        allowOrigins: ["*"]
+    }
+}
+
+service on new http:Listener(9090) {
```
Comment (NipunaRanasinghe) [inline openapi-cli/src/test/resources/ballerina-to-openapi/response/service_config_with_cors_without_base_path.bal:9]: Additional newline

**Review by NipunaRanasinghe — APPROVED** [review]
LGTM

**Review by aneeshafedo — APPROVED** [review]

**Pull request merged**